src/docfx/restore/RestoreFile.cs

```diff
@@ -114,6 +68,14 @@ public static IEnumerable<string> GetFileReferences(this Config config)
             yield return config.MonikerDefinition;
         }
 
+        public static string GetRestorePath(string url)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(url));
+            Debug.Assert(HrefUtility.IsHttpHref(url));
+
+            return PathUtility.NormalizeFile(Path.Combine(AppData.GetFileDownloadDir(url), HashUtility.GetMd5HashShort(url)));
```

Comment:
`GetFileDownloadDir` already contains url hash, so no need for `HashUtility.GetMd5HashShort(url)`

Reply:
AppData.GetFileDownloadDir(url) is the dir of this file, I like @superyyrrzz 's idea, `content` and `etag`